static/elements/chromedash-userlist.js

```diff
@@ -0,0 +1,90 @@
+import {LitElement, html} from 'https://unpkg.com/@polymer/lit-element@latest/lit-element.js?module';
+
+class ChromedashUserlist extends LitElement {
+  static get properties() {
+    return {
+      actionPath: {type: String},
+      users: {type: Array},
+    };
+  }
+
+  constructor() {
+    super();
+    this.users = [];
+  }
+
+  addUser(user) {
+    this.users.splice(0, 0, user);
+    this.users = this.users.slice(0); // Refresh the list
+  }
+
+  removeUser(idx) {
+    this.users.splice(idx, 1);
+    this.users = this.users.slice(0); // Refresh the list
+  }
+
+  ajaxSubmit(e) {
+    e.preventDefault();
+    const formEl = this.shadowRoot.querySelector('form');
+
+    if (formEl.checkValidity()) {
+      const email = formEl.querySelector('input[name="email"]').value;
+      const formData = new FormData();
+      formData.append('email', email);
+
+      fetch(this.actionPath, {
+        method: 'POST',
+        body: formData,
+        credentials: 'same-origin', // Needed for admin permissions to be sent.
+      }).then((resp) => {
+        if (resp.status === 200) {
+          alert('Thanks. But that user already exists');
+          throw new Error('User already added');
+        } else if (resp.status !== 201) {
+          throw new Error('Sever error adding new user');
+        }
+        return resp.json();
+      }).then((json) => {
+        this.addUser(json);
+        formEl.reset();
+      });
+    }
+  }
+
+  ajaxDelete(e) {
+    e.preventDefault();
+    if (!confirm('Remove user?')) {
+      return;
+    }
+
+    const idx = e.target.dataset.index;
+
+    fetch(e.currentTarget.href, {
```

Comment:
If you make this an async function and use await, I think fetch errors will propagate to the console